File: SameTree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
        
class Solution:
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        if p==None and q==None:
            return True
        if p==None or q==None:
            return False
        if p.val!=q.val:
            return False
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
        
if __name__=='__main__':
    p = TreeNode(1);
    a = TreeNode(2);
    q = TreeNode(1);
    b = TreeNode(2);
    
    p.left = a
    p.right = b
    
    solution = Solution()
    print(solution.isSameTree(p, q))

File: SameTree.cpp
#include<iostream>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(p!=NULL&&q!=NULL){
            if(p->val!=q->val) return false;
            else{
                return isSameTree(p->left, q->left)&&isSameTree(p->right, q->right);
            }
        }else{
            if(p==NULL&&q==NULL) return true;
            else return false;
        }
    }
};

int main(){
    TreeNode *p=new TreeNode(1);
    TreeNode *a=new TreeNode(2);
    TreeNode *q=new TreeNode(1);
    TreeNode *b=new TreeNode(2);
    
    p->left=a;
    q->right=b;
    
    Solution *solution=new Solution();
    solution->isSameTree(p, q) ? cout<<"True" : cout<<"False";
    
    return 0;
}
